src/components/ToDoForm.js

```diff
@@ -1,20 +1,35 @@
-import React from 'react'
+import { React, useState } from 'react'
 
-export default function ToDoForm() {
+export default function ToDoForm( props) {
+    const [newItem, setNewItem] = useState("")
 
-  function handleSubmit(event) {
-    event.preventDefault();
-  }
+    function handleSubmit(event) {
+        event.preventDefault();
+        if (newItem === "") 
+            return
+        props.onSubmit(newItem);
+        setNewItem("");
+    };
 
-  return (
-    <form className="todo-form" onSubmit={ handleSubmit }>
-        <input 
-        type="text" 
-        className="todo-form__input" 
-        placeholder="Add new item"/>
-        <button 
-        type="submit" 
-        className="todo-form__button">add</button>
-    </form>
-  )
+    function handleInputChange(event) {
+        setNewItem(event.target.value)
+    };
+
+    return (
+        <>
+            <form className="todo-form" onSubmit={ handleSubmit }>
+                <label htmlFor="item">New item</label>
+                <input
+                    type="text"
+                    id="item"
+                    className="todo-form__input" 
+                    value={newItem}
+                    onChange={event => handleInputChange(event)}
```

Comment:
onChange={handleInputChange}
dzięki temu unikniesz tworzenia niepotrzebnej arrow function. Tu nie ma to wielkiego znaczenia, bo wpływ jest minimalny, ale warto wiedzieć:
https://stackoverflow.com/questions/48699573/correct-use-of-arrow-functions-in-react